Nitrox.Launcher/Nitrox.Launcher.csproj

```diff
@@ -23,16 +28,60 @@
-        <PackageReference Condition="'$(Configuration)' == 'Debug'" Include="Avalonia.Diagnostics" Version="0.10.18" />
-        <PackageReference Include="Avalonia.ReactiveUI" Version="0.10.18" />
+        <PackageReference Condition="'$(Configuration)' == 'Debug'" Include="Avalonia.Diagnostics" Version="$(AvaloniaVersion)" />
+        <PackageReference Include="Avalonia.ReactiveUI" Version="$(AvaloniaVersion)" />
         <PackageReference Include="Markdown.Avalonia" Version="0.10.12" />
         <PackageReference Include="ReactiveUI.Validation" Version="3.0.1" />
         <PackageReference Include="XamlNameReferenceGenerator" Version="1.3.4" />
     </ItemGroup>
     <ItemGroup>
         <UpToDateCheckInput Remove="Assets\Images\**" />
     </ItemGroup>
+
+    <ItemGroup>
+        <ProjectReference Include="..\NitroxModel\NitroxModel.csproj" />
+        <ProjectReference Include="..\NitroxServer\NitroxServer.csproj" />
+        <ProjectReference Condition="$(BuildSubnautica)==true" Include="..\NitroxServer-Subnautica\NitroxServer-Subnautica.csproj" />
+    </ItemGroup>
+
+    <Target Name="BuildSubnautica" AfterTargets="Build" Condition="$(BuildSubnautica)==True">
+        <ItemGroup>
+            <ProjectToBuildSN Include="..\NitroxModel\NitroxModel.csproj">
+                <AdditionalProperties>TargetFramework=net472</AdditionalProperties>
+            </ProjectToBuildSN>
+            <ProjectToBuildSN Include="..\NitroxModel-Subnautica\NitroxModel-Subnautica.csproj">
+
+            </ProjectToBuildSN>
+            <ProjectToBuildSN Include="..\NitroxClient\NitroxClient.csproj">
+
+            </ProjectToBuildSN>
+            <ProjectToBuildSN Include="..\NitroxPatcher\NitroxPatcher.csproj">
+
+            </ProjectToBuildSN>
+        </ItemGroup>
+        
+        <!-- Builds the Nitrox Clientside libraries for Subnautica -->
+        <!-- OutputPath | Sets the output path to lib -->
+        <!-- SatelliteResourceLanguages | Gets rid of useless localization folders in lib -->
+        <MSBuild
+            Projects="@(ProjectToBuildSN)"
+            Properties="OutputPath=$(TargetDir)\lib;SatelliteResourceLanguages=en">
+        </MSBuild>
+        
+        <ItemGroup>
+            <NitroxSubnauticaAssets Include="..\Nitrox.Assets.Subnautica\AssetBundles\**\*." />
+        </ItemGroup>
+        <ItemGroup>
+            <NitroxSubnauticaStaticDlls Include="..\Nitrox.Assets.Subnautica\**\*.dll" />
+        </ItemGroup>
+        <ItemGroup>
+            <NitroxSubnauticaLanguageFiles Include="..\Nitrox.Assets.Subnautica\LanguageFiles\*.json" />
+        </ItemGroup>
+        <Copy SourceFiles="@(NitroxSubnauticaAssets)" DestinationFolder="$(TargetDir)\AssetBundles\%(RecursiveDir)" />
+        <Copy SourceFiles="@(NitroxSubnauticaStaticDlls)" DestinationFolder="$(TargetDir)\lib\%(RecursiveDir)" />
+        <Copy SourceFiles="@(NitroxSubnauticaLanguageFiles)" DestinationFolder="$(TargetDir)\LanguageFiles\%(RecursiveDir)" />
+    </Target>
```

Comment:
The only way to fix these would be to use net472 for the launcher, or port most Nitrox.* projects to net6, which isn't possible due to these being required in Unity-side code. We could also decouple building Nitrox's clientside libs and the libs required for the launcher, but this would result in two DLLs with identical functionality.

Reply:
Separating game and server DLLs is indeed the way to go. The game has some support for netstandard 2.0 with latest subnautica update (living large). But not all code is compatible (Serilog causes a `TypeLoadException` from my testing).

Reply:
> The servers could be built separately and then the launcher would just run the exe.
This is how we want to do it. Start the exe with some arguments/environment-variables, and then never interface with server again other than packet serialization.

I'm in the works with splitting client and server up and be build separately. But the serializer (BinaryPack) we use for packets doesn't work with netstandard 2.0 which is currently blocking me.

If you keep the .csproj changes for later I would appreciate it.

Reply:
Will hold off the csproj changes.